File: YuGiOh2/Cards/C54109233.cs
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 救世主之蚁地狱
    /// </summary>
    public class C54109233
    {
        public static int Type { get; } = (int)AffectMomentType.WhenSummoned;

        public static void ProcessEffect(string targetID, Player player)
        {
            if (targetID == null)
                return;

            for (int i = 0; i < 5; i++)
            {
                if (player.Enemy.Field.MonsterFields[i] == null ||
                    player.Enemy.Field.MonsterFields[i].UID != targetID ||
                    player.Enemy.Field.MonsterFields[i].Level > 4)
                    continue;

                player.Enemy.AddCardToGrave(player.Enemy.Field.MonsterFields[i]);
                player.Enemy.Field.MonsterFields[i] = null;
            }
        }
    }
}

File: YuGiOh2/Scripts/C54109233.lua
Type = 1;
--救世主之蚁地狱
function processEffect(targetID, player)
    if targetID == nil then
        return ;
    end
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and monster.UID == targetID and monster.Level <= 4 then
            player.Enemy:AddCardToGrave(monster);
            player.Enemy:ClearMonsterField(i - 1);
        end
    end
end